benchmarks/tpcc/src/main/java/com/google/cloud/pgadapter/tpcc/JavaClientBenchmarkRunner.java

```diff
@@ -0,0 +1,257 @@
+  }
+
+  Object[] paramQueryRow(String sql, Object[] params) throws SQLException {
+    ParametersInfo parametersInfo = convertPositionalParametersToNamedParameters(sql, '?', "$");
+    Statement.Builder builder = Statement.newBuilder(parametersInfo.sqlWithNamedParameters);
+    bindParams(builder, params);
+    Statement stmt = builder.build();
+
+    Stopwatch stopwatch = Stopwatch.createStarted();
+    ResultSet resultSet = transactionThreadLocal.get().executeQuery(stmt);
+    Duration executionDuration = stopwatch.elapsed();
+    metrics.recordLatency(executionDuration.toMillis());
+
+    if (!resultSet.next()) {
+      throw new RowNotFoundException(String.format("No results found for: %s", sql));
+    }
+
+    Object[] result = new Object[resultSet.getColumnCount()];
+    for (int i = 0; i < result.length; i++) {
+      result[i] = getObject(resultSet.getValue(i));
+    }
+    return result;
+  }
+
+  void executeParamStatement(String sql, Object[] params) throws SQLException {
+    ParametersInfo parametersInfo = convertPositionalParametersToNamedParameters(sql, '?', "$");
+    Statement.Builder builder = Statement.newBuilder(parametersInfo.sqlWithNamedParameters);
+    bindParams(builder, params);
+    Statement stmt = builder.build();
+
+    Stopwatch stopwatch = Stopwatch.createStarted();
+    transactionThreadLocal.get().executeUpdate(stmt);
+    Duration executionDuration = stopwatch.elapsed();
+    metrics.recordLatency(executionDuration.toMillis());
+  }
+
+  public static class ParametersInfo {
+    public final int numberOfParameters;
+    public final String sqlWithNamedParameters;
+
+    ParametersInfo(int numberOfParameters, String sqlWithNamedParameters) {
+      this.numberOfParameters = numberOfParameters;
+      this.sqlWithNamedParameters = sqlWithNamedParameters;
+    }
+  }
+
+  public ParametersInfo convertPositionalParametersToNamedParameters(
+      String sql, char paramChar, String queryParameterPrefix) {
+    final String namedParamPrefix = queryParameterPrefix;
+    StringBuilder named = new StringBuilder(sql.length() + countOccurrencesOf(paramChar, sql));
+    int index = 0;
+    int paramIndex = 1;
+    while (index < sql.length()) {
+      char c = sql.charAt(index);
+      if (c == paramChar) {
+        named.append(namedParamPrefix).append(paramIndex);
+        paramIndex++;
+        index++;
+      } else {
+        named.append(c);
+        index++;
+      }
+    }
+    return new ParametersInfo(paramIndex - 1, named.toString());
+  }
+
+  /** Convenience method that is used to estimate the number of parameters in a SQL statement. */
+  static int countOccurrencesOf(char c, String string) {
+    int res = 0;
+    for (int i = 0; i < string.length(); i++) {
+      if (string.charAt(i) == c) {
+        res++;
+      }
+    }
+    return res;
+  }
+
+  // If getAsObject() in Struct can be a public method, then we should use it
+  // instead of using the following approach.
+  Object getObject(Value value) {
```

Comment:
Instead of relying on exception handling, can't you just call `Value#getType()` and use that in a switch statement to get the correct type?

Reply:
Good point. I didn't know `Value#getType()`. Cannot use switch as it requires `constant expression required`. Updated.

Reply:
(Note for the future: You can call `value.getType().getTypeCode()` and use that in a switch statement)